File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        int res = 0;

        for (int num: nums) {
            set.add(num);
        }

        for (int num: nums) {
            if (!set.contains(num - 1)) {
                int streakLength = 1;
                while (set.contains(num + 1)) {
                    streakLength++;
                    num+=1;
                }
                res = Math.max(res, streakLength);
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/maximum-subarray/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    maxSubArray(nums) {
        let maxSub = nums[0]
        let currSum = 0

        for (let i = 0; i < nums.length; i++) {
            if (currSum < 0) {
                currSum = 0
            } 

            currSum += nums[i]
            maxSub = Math.max(currSum, maxSub)
        }

        return maxSub
    }
}
